src/objects/zcl_abapgit_object_odso.clas.abap

```diff
@@ -0,0 +1,363 @@
+CLASS zcl_abapgit_object_odso DEFINITION
+  PUBLIC
+  INHERITING FROM zcl_abapgit_objects_super
+  FINAL
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+
+    INTERFACES zif_abapgit_object .
+
+    ALIASES mo_files
+      FOR zif_abapgit_object~mo_files .
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+
+    METHODS:
+      clear_field
+        IMPORTING
+          iv_fieldname TYPE string
+        CHANGING
+          cs_metadata  TYPE any.
+ENDCLASS.
+
+
+CLASS zcl_abapgit_object_odso IMPLEMENTATION.
+
+
+  METHOD zif_abapgit_object~changed_by.
+
+    DATA: lv_dsonam  TYPE c LENGTH 30,
+          ls_return  TYPE bapiret2,
+          lr_details TYPE REF TO data.
+
+    FIELD-SYMBOLS: <ls_details> TYPE any,
+                   <lv_tstpnm>  TYPE any.
+
+    TRY.
+        CREATE DATA lr_details TYPE ('BAPI6116').
+      CATCH cx_sy_create_data_error.
+        zcx_abapgit_exception=>raise( |ODSO is not supported on this system| ).
+    ENDTRY.
+
+    ASSIGN lr_details->* TO <ls_details>.
+
+    lv_dsonam = ms_item-obj_name.
+
+    CALL FUNCTION 'BAPI_ODSO_GETDETAIL'
+      EXPORTING
+        odsobject = lv_dsonam
+      IMPORTING
+        details   = <ls_details>
+        return    = ls_return.
+
+    IF ls_return-type = 'E'.
+      zcx_abapgit_exception=>raise( |Error when geting changed by of ODSO: { ls_return-message }| ).
+    ENDIF.
+
+    ASSIGN COMPONENT 'TSTPNM' OF STRUCTURE <ls_details> TO <lv_tstpnm>.
+
+    rv_user = <lv_tstpnm>.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lv_odsonam    TYPE c LENGTH 30,
+          lv_objname    TYPE sobj_name,
+          lo_collection TYPE REF TO object,
+          lt_msg        TYPE rs_t_msg,
```

Comment:
`RS_T_MSG` is part of BW. As compatibility issues are likely I'd suggest to use `STANDARD TABLE OF BAL_S_MSG` instead.

Reply:
fixed

Reply:
I have to try to automatize somehow this kind of check.